src/Modules/missing_functionality.jl

```diff
@@ -77,3 +77,114 @@ end=#
     
 subquo_type(::Type{RingType}) where {RingType<:Ring} = SubquoModule{elem_type(RingType)}
 subquo_type(R::RingType) where {RingType<:Ring} = subquo_type(typeof(R))
+
+function _kbase(I::SubModuleOfFreeModule{<:MPolyRingElem{<:FieldElem}};
+    ordering::ModuleOrdering=default_ordering(ambient_free_module(I))
+  )
+  lead_I = leading_module(I, ordering)
+  @show ordering
+  @show lead_I
+  F = ambient_free_module(I)
+  R = base_ring(F)
+  # Iterate through all components until we have everything
+  result = elem_type(F)[]
+  for i in 1:ngens(F)
+    m = ideal(R, gens(R))
+    pom = ideal(R, one(R))
+    while true
+      new_mons = [x for x in gens(pom) if !(x*F[i] in lead_I)]
+      iszero(length(new_mons)) && break
+      result = vcat(result, [x*F[i] for x in new_mons])
```

Comment:
This is an extremely minor thing, but I just wanted to point out that depending on the ring, `F[i]` might allocate a new object each time it is called, so perhaps pull it out before the loop? Also, `x*F[i]` is computed twice for all remaining in `new_mons`. 
```suggestion
    Fi = F[i]
    m = ideal(R, gens(R))
    pom = ideal(R, one(R))
    while true
      new_mons = [x*Fi for x in gens(pom)]
      filter!(x -> !(x in lead_I), new_mons)
      ismpty(new_mons) && break
      result = vcat(result, new_mons)
```